presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2141,6 +2141,27 @@ public void testJoinWithLessThanOnDatesInJoinClause()
     }
 
     @Test
+    public void testJoinWithNonDeterministicLessThan()
+    {
+        MaterializedRow actualRow = getOnlyElement(computeActual(
```

Comment:
I think this code is ok. If we combine the queries, the optimizer could theoretically mangle the plan to the point where we're not testing what we want.